### ISSUE TITLE
Cabal project support

### ISSUE THREAD

**Issue by Fresheyeball** [commercialhaskell/intero#493]
Since intero is just a ghci process (or so is my understanding), I don't see why it needs to rely on stack necessarily. It would be very nice to use intero with plain old cabal projects.

**Comment by joehillen** [comment]
Try [Dante](https://github.com/jyp/dante)

**Comment by Fresheyeball** [comment]
@joehillen I use Dante now. It's ok. Would still rather see intero, since there is much much more built on intero. And Dante only works with emacs.

**Comment by mgsloan** [comment]
The intero executable does not itself rely on stack, currently.

The only bits that rely on stack is in the elisp - the procedure to reliably install intero built against the right version of ghc, and the bits to invoke intero with the correct flags.  I do not know of a good way to do this with cabal.  One issue is, AFAIK cabal repl has no way to load multiple packages.  I believe that stack ghci is quite a lot better for the intero usecase.

So, one consideration with adding cabal support is it could mar intero's attempt at "it just works" type behavior (not saying it's been achieved quite yet).

Not sure what @chrisdone thinks about it, but personally, I'd be fine merging cabal project that:

1. Opt in

2. Could reliably install intero (tricky considering the wild west of cabal!  in a nix setting it could work out)

3. Did not place undue maintenance burden / complication on the rest of the elisp.

4. Cabal / nix users pledged to maintain it.  The maintainers of intero do not typically use cabal, so we have little incentive to work on this.

An alternative approach here would just be to make stack work better with nix.  We also do not typically use nix, preferring things like docker images for truly reproducible environments.

Though, recently I am starting to try to use it on a client project.  If I learn enough about nix there, I might work on dialing in stack's nix support.  It's a bit rough going, though, something is very wrong with the binary caches. I got kinda frustrated when it started doing an ocaml build to rebuild `flow`, I think it also built `R` from scratch, I saw it using gfortran for that.  Took a /long/ time to boot the environment, far longer than it would have taken to download and use a docker image.  I don't really like using gentoo-like systems, but that's personal preference, and perhaps the binary caching issue can be fixed.

So yeah, I wouldn't be opposed to improving integrations here, though I won't speak for @chrisdone and he's the primary author / maintainer here.  If it's going to happen, then cabal / nix users are going to need to step up to the plate and make it happen.  It doesn't make much sense for us to build integration with tools that we don't use.
